sensors/triggers/kafka/kafka.go

```diff
@@ -211,3 +230,51 @@ func (t *KafkaTrigger) Execute(ctx context.Context, events map[string]*v1alpha1.
 func (t *KafkaTrigger) ApplyPolicy(ctx context.Context, resource interface{}) error {
 	return nil
 }
+
+// getSchemaFromRegistry returns a schema from registry
+func getSchemaFromRegistry(sr *apicommon.SchemaRegistryConfig) (*srclient.Schema, error) {
+	user, err := common.GetSecretFromVolume(sr.UserSecret)
+	if err != nil {
+		return nil, fmt.Errorf("error getting user value from secret, %w", err)
+	}
+	password, err := common.GetSecretFromVolume(sr.PasswordSecret)
+	if err != nil {
+		return nil, fmt.Errorf("error getting password value from secret, %w", err)
+	}
+
+	schemaRegistryClient := srclient.CreateSchemaRegistryClient(sr.URL)
+	schemaRegistryClient.SetCredentials(user, password)
+
+	schema, err := schemaRegistryClient.GetSchema(int(sr.SchemaId))
```

Comment:
rather than have id as a value in config, which is likely to be different per environment, instead could the id be found on the message itself? like https://github.com/nats-io/nats-kafka/blob/9e878e16ee0a378eb9d23cd41ee2df8e406ece40/server/kafka/consumer.go#L320-L330

Reply:
I do not agree with this point.

This means: All payloads/messages must have schemaID.
If we have a fixed wehbook source like Gitlab where we can't edit the payload/messages then all messages need to be edited to add schemaId.

Reply:
@Thomasr-02 if schemaId not supplied in the config then read schemaid off the message

Reply:
I understood. 

I believe this can be an improvement, it doesn't fit the scope of the solution I proposed